tola_management/programadmin.py

```diff
@@ -417,21 +414,118 @@ class ProgramAdminViewSet(viewsets.ModelViewSet):
     pagination_class = Paginator
     permission_classes = [permissions.IsAuthenticated, HasProgramAdminAccess]
 
+    @classmethod
+    def base_queryset(cls):
+        """this adds annotations for the "users" and "organizations" links using annotations
+        
+            For performance reasons, looking up every user with permission to see the program individually for
+            all 20+ in a paginated set was costly.  This annotates the information in one (admittedly spendy) query
+        """
+        # start with the correctly annotated-for-rf queryset:
+        queryset = Program.rf_aware_objects.all()
+        queryset = queryset.annotate(
+            # this counts users who are not mercy corps users (partner access) who have been assigned:
+            program_access_users_count=models.functions.Coalesce( # coalesce so None is 0 (summable later)
+                models.Subquery(
+                    ProgramAccess.objects.filter(
+                        program=models.OuterRef('pk')
+                    ).exclude(
+                        tolauser__organization_id=Organization.MERCY_CORPS_ID
+                    ).order_by().values('program').annotate(
+                        users_count=models.Count('tolauser', distinct=True)
+                    ).values('users_count')[:1],
+                    output_field=models.IntegerField()
+                ), 0
+            ),
+            # only mercy corps users can be given country-wide access, so count them with filters to minimize joins:
+            country_access_users_count=models.functions.Coalesce( # coalesce so None is 0 (summable later)
+                models.Sum( # sum because there may be multiple countries, each with a user count
+                    models.Subquery(
+                        CountryAccess.objects.filter(
+                            country=models.OuterRef('country'),
```

Comment:
Curious why this isn't `country__in=models.OuterRef('country')`, since Program.country is M2M.

Reply:
I found that the m2m join caused a summing problem - I'm not 100% sure why (m2m joins and annotations have unpredictable effects) so I went with country= and then a Sum, so what it's doing is counting the number of country users _per country_ and then adding them up for all the countries in the program.  That way it gets the right results - I'm not sure if there's a better way to do it?